crates/bevy_ecs/src/change_detection.rs

```diff
@@ -428,6 +428,22 @@ macro_rules! impl_methods {
                 }
             }
 
+            /// Optionally maps to an inner value by applying a function to the contained reference.
+            /// This is useful in a situation where you need to convert a `Mut<T>` to a `Mut<U>`, but only if `T` contains `U`.
+            ///
+            /// As with `map_unchanged`, you should never modify the argument passed to the closure.
+            pub fn filter_map_unchanged<U: ?Sized>(self, f: impl FnOnce(&mut $target) -> Option<&mut U>) -> Option<Mut<'w, U>> {
+                match f(self.value) {
+                    Some(value) => Some(Mut {
+                        value,
+                        ticks: self.ticks,
+                        #[cfg(feature = "track_change_detection")]
+                        changed_by: self.changed_by,
+                    }),
+                    None => None
+                }
```

Comment:
I think you could simplify the `match` using `Option::map`, like

```suggestion
                let value = f(self.value);
                value.map(|value| Mut {
                    value,
                    ticks: self.ticks,
                    #[cfg(feature = "track_change_detection")]
                    changed_by: self.changed_by,
                })
```